repository-ydb-v2/src/main/java/tech/ydb/yoj/repository/ydb/yql/YqlPredicate.java

```diff
@@ -76,6 +76,16 @@ public static void setUseLegacyRel(boolean value) {
         YqlPredicate.useLegacyRel.set(value);
     }
 
+    public static YqlPredicate from(Collection<? extends YqlStatementPart<?>> parts) {
+        YqlPredicate predicate = YqlPredicate.alwaysTrue();
```

Comment:
This method will generate a bit better YQL code if you use `YqlPredicate.and(Collection<YqlPredicate>)` on a collection of `parts` that are `instanceof YqlPredicate`, instead instead of turning the `YqlStatement.predicateFrom()`'s `Stream::reduce` into a `for` loop. (The `YqlPredicate.and()` method also returns `YqlPredicate.alwaysTrue()` if the predicate collection is empty, so all the invariants of previous implementations would be satisfied, I think.)